cmd/ddev/cmd/exec_test.go

```diff
@@ -109,10 +113,37 @@ func TestCmdExec(t *testing.T) {
 	assert.NoError(err)
 	assert.NotEqual("/var/www/html", strings.TrimSpace(out))
 
-	// Complex commands with spaces should also work, even without the --raw flag
-	out, err = exec.RunHostCommand(DdevBin, "exec", "bash", "-c", "for i; do echo $i; done", "_", "string with spaces")
+	// Create a bash script for testing
+	_, err = exec.RunHostCommand(DdevBin, "exec", "echo 'for i; do echo $i; done' > echo_args.sh")
+	assert.NoError(err)
+	err = app.MutagenSyncFlush()
+	assert.NoError(err)
+	assert.FileExists(filepath.Join(site.Dir, "echo_args.sh"))
+
+	// Arguments with spaces should not be split, quotes should be preserved
+	out, err = exec.RunHostCommand(DdevBin, "exec", "bash", "echo_args.sh", "string with \"quotes\" and spaces", "another 'quoted' string")
+	assert.NoError(err)
+	assert.Equal("string with \"quotes\" and spaces\nanother 'quoted' string", strings.TrimSpace(out))
+
+	// Bash expansion should work
+	out, err = exec.RunHostCommand(DdevBin, "exec", "echo", "$IS_DDEV_PROJECT")
+	assert.NoError(err)
+	assert.Equal("true", strings.TrimSpace(out))
+
+	// Bash expansion should work (using arg with spaces)
+	out, err = exec.RunHostCommand(DdevBin, "exec", "echo", "$IS_DDEV_PROJECT\n${DDEV_NON_EXISTING_TEST_VARIABLE:-foobar}")
+	assert.NoError(err)
+	assert.Equal("true\nfoobar", strings.TrimSpace(out))
+
+	// Bash expansion doesn't work with --raw, it's expected
+	out, err = exec.RunHostCommand(DdevBin, "exec", "--raw", "echo", "$IS_DDEV_PROJECT")
+	assert.NoError(err)
+	assert.Equal("$IS_DDEV_PROJECT", strings.TrimSpace(out))
+
+	// Bash expansion doesn't work with --raw, it's expected (using arg with spaces)
+	out, err = exec.RunHostCommand(DdevBin, "exec", "--raw", "echo", "$IS_DDEV_PROJECT\n${DDEV_NON_EXISTING_TEST_VARIABLE:-foobar}")
 	assert.NoError(err)
-	assert.Equal("string with spaces", strings.TrimSpace(out))
+	assert.Equal("$IS_DDEV_PROJECT\n${DDEV_NON_EXISTING_TEST_VARIABLE:-foobar}", strings.TrimSpace(out))
```

Comment:
This is a bit baffling, so a comment explaining why it's here would be an improvement.

Reply:
Updated with comment.